src/algebraic_topology/fundamental_groupoid/retraction.lean

```diff
@@ -42,56 +42,55 @@ section unbundled
 
 /-- We define `A ⊆ X` to be a topological subspace by defining the property `A_filter` picking
 elements of `A` out of `X`. This inherits the topology on `X` via `subtype.topological_space`. -/
-variables {X : Type*} {A_filter : X → Prop} [topological_space X]
+variables {X : Type*} {A : set X} [topological_space X]
 
-def inclusion (X : Type*) (A_filter : X → Prop) [topological_space X] :
-  C(subtype A_filter, X) := ⟨subtype.restrict A_filter id⟩
+def inclusion (X : Type*) (A : set X) [topological_space X] :
+  C(A, X) := ⟨set.restrict A id⟩
 
 /-- A continuous map `r : X → A` for `A ⊆ X` is a topological retraction when it is the identity
 when restricted to A. -/
-structure is_top_retraction (r : X → subtype A_filter) extends continuous r : Prop :=
-(id_of_retraction_of_inclusion : r ∘ (inclusion X A_filter) = id)
+structure is_top_retraction (r : X → A) extends continuous r : Prop :=
+(id_of_retraction_of_inclusion : r ∘ (inclusion X A) = id)
 
 /-- A topological retraction is continuous by definition. -/
-lemma is_top_retraction.continuous {r : X → subtype A_filter} (hr : is_top_retraction r) :
-  continuous r :=
-hr.to_continuous
+lemma is_top_retraction.continuous {r : X → A} (hr : is_top_retraction r) :
+  continuous r := hr.to_continuous
 
 end unbundled
 
 
 /-- The bundled form of `is_top_retraction`. -/
-structure top_retraction (X : Type*) (A_filter : X → Prop) [topological_space X] :=
-(to_fun : X → subtype A_filter)
+structure top_retraction (X : Type*) (A : set X) [topological_space X] :=
+(to_fun : X → A)
 (top_retraction' : is_top_retraction to_fun)
 
 
 namespace top_retraction
 
-variables {X : Type*} {A_filter : X → Prop} [topological_space X]
+variables {X : Type*} {A : set X} [topological_space X]
 
 /-- Helper to coerce a topological retraction to a continuous map. -/
-def to_continuous_map (r : top_retraction X A_filter) : C(X, subtype A_filter) :=
+def to_continuous_map (r : top_retraction X A) : C(X, subtype A) :=
```

Comment:
> Thanks for adopting my suggestions; here are just some comments on the code. For the mathematics involved, I'm not sure but it would be better if you develop more theory around topological retractions to justify the introduction of a new structure.

It's in the works :) - I'm working on building up the background for the Brouwer fixed point theorem, and this is one of the ingredients for that.